### PR TITLE
WARN: Catch warning in tests

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke

### CODE REVIEW

File: pandas/tests/plotting/frame/test_frame_color.py [hunk 1]
```diff
@@ -204,7 +206,11 @@ def test_scatter_with_c_column_name_with_colors(self, cmap, kw):
             columns=["length", "width"],
         )
         df["species"] = ["r", "r", "g", "g", "b"]
-        ax = df.plot.scatter(x=0, y=1, cmap=cmap, **{kw: "species"})
+        if mpl_ge_3_6_0() and cmap is not None:
+            with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
```
Comment (jbrockmendel) [inline pandas/tests/plotting/frame/test_frame_color.py:210]: does this indicate that we are doing something wrong/outdated?

Comment (mroeschke) [reply pandas/tests/plotting/frame/test_frame_color.py:210]: @MarcoGorelli looks like you raised and addressed this issue in https://github.com/pandas-dev/pandas/pull/34344

Do you remember if passing `cmap` in `plot.scatter was necessary` to test this behavior?

Comment (mroeschke) [reply pandas/tests/plotting/frame/test_frame_color.py:210]: FWIW: I think this is a new warning in mpl 3.6 so `plot.scatter` may have been doing something "wrong" before?

Comment (phofl) [reply pandas/tests/plotting/frame/test_frame_color.py:210]: It was silently ignored before. They added stricter checks in a few places. If we want to check that the colormap is not applied, then we have to check for the user warning

**Review by MarcoGorelli — COMMENTED** [review]
Looks good to me

Not overly keen on `if-then` inside tests, slight preference to split the test into two:
- `test_scatter_with_c_column_name_with_colors_no_cmap`
- `test_scatter_with_c_column_name_with_colors_with_cmap`

and also to add a comment that the warning comes from matplotlib, so it's fine to not check the stacklevel

Not blockers though

File: pandas/tests/plotting/frame/test_frame_color.py [hunk 2]
```diff
@@ -15,6 +15,8 @@
     _check_plot_works,
 )
 
+from pandas.plotting._matplotlib.compat import mpl_ge_3_6_0
```
Comment (mroeschke) [inline pandas/tests/plotting/frame/test_frame_color.py:18]: I think this needs to be imported in the test

Comment (phofl) [reply pandas/tests/plotting/frame/test_frame_color.py:18]: thx, moved

**Comment by phofl** [comment]
I'd prefer keeping it as one test, this makes comparing easier.

**Review by MarcoGorelli — APPROVED** [review]
looks good to me

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**